src/ngraph/runtime/cpu/pass/cpu_fusion.cpp

```diff
@@ -1272,3 +1275,105 @@ void ngraph::runtime::cpu::pass::CPUFusion::construct_bounded_relu()
     auto m = std::make_shared<pattern::Matcher>(min, callback);
     this->add_matcher(m);
 }
+
+void ngraph::runtime::cpu::pass::CPUFusion::construct_horizontal_fusion()
+{
+    auto is_concat_or_maxpool = [](std::shared_ptr<Node> n) {
+        return std::dynamic_pointer_cast<ngraph::op::Concat>(n) ||
+               std::dynamic_pointer_cast<ngraph::op::MaxPool>(n);
+    };
+
+    auto data_conv = std::make_shared<pattern::op::Label>(
+        element::f32, Shape{1, 256, 35, 35}, is_concat_or_maxpool);
+    auto filters = std::make_shared<pattern::op::Label>(element::f32, Shape{64, 256, 1, 1});
+    auto bias = std::make_shared<pattern::op::Label>(element::f32, Shape{64});
+
+    auto conv_bias = std::make_shared<ngraph::op::ConvolutionBias>(data_conv,
+                                                                   filters,
+                                                                   bias,
+                                                                   Strides{1, 1},
+                                                                   Strides{1, 1},
+                                                                   CoordinateDiff{0, 0},
+                                                                   CoordinateDiff{0, 0},
+                                                                   Strides{1, 1},
+                                                                   true);
+
+    pattern::graph_rewrite_callback callback = [data_conv](pattern::Matcher& m) {
+        NGRAPH_DEBUG << "In a callback for inception horizontal fusion for "
+                     << m.get_match_root()->get_name();
+
+        auto conv_bias_root = std::dynamic_pointer_cast<op::ConvolutionBias>(m.get_match_root());
+
+        //check if the node has been replaced
+        if (conv_bias_root->get_users().empty())
+        {
+            return false;
+        }
+
+        //check if 1x1 filter
+        auto m_filters_shape = conv_bias_root->get_input_shape(1);
```

Comment:
I thought we could always do this optimization regardless of a filter size as long as all convolutions are using the same data?